spec/struct.dd

```diff
@@ -70,11 +72,21 @@ $(H2 $(LNAME2 struct_layout, Struct Layout))
     $(OL
     $(LI The default layout of the fields of a struct is an exact
     match with the $(I associated C compiler).)
+    $(LI g++ and clang++ differ in how empty structs are handled. Both return `1` from `sizeof`,
+    however, clang++ does not push them onto the parameter stack while g++ does. This is a
+    binary incompatibility between g++ and clang++.
+    dmd follows clang++ behavior for OSX and FreeBSD, and g++ behavior for Linux and other
+    Posix platforms.
```

Comment:
I suspect that in the long run, it'll make more sense to use clang's behavior on pretty much everything other than Linux or Linux-based systems such as Android (though Android may prefer clang too; I'm not sure; it's not supported by dmd at the moment though, so it doesn't really matter for dmd). Certainly, as I understand it, all of the BSDs are moving towards ditching gcc in favor of clang, but right now, dmd isn't officially supporting any of those platforms, so it isn't an issue at the moment, and this can be changed to take such systems into account if they ever do end up with dmd support. And given how clang supposedly tries to be compatible with gcc, I wonder if clang is ever going to change what it's doing to match gcc. The whole thing is ugly either way, and we can cross that bridge if it ever happens.

Reply:
The plan is to match the behavior of the default compiler on the target platform, whether that is clang or gcc.